map.go

```diff
@@ -391,6 +404,36 @@ func (m *Map[K, V]) doCompute(
 		h2w := broadcast(h2)
 		bidx := uint64(len(table.buckets)-1) & h1
 		rootb := &table.buckets[bidx]
+
+		if loadExitFlag != 0 {
+			b := rootb
+		load:
+			for {
+				metaw := b.meta.Load()
+				markedw := markZeroBytes(metaw^h2w) & metaMask
+				for markedw != 0 {
+					idx := firstMarkedByteIndex(markedw)
+					e := b.entries[idx].Load()
+					if e != nil {
+						if e.key == key {
+							if loadExitFlag == 1 {
+								return e.value, true
+							}
+							break load
+						}
+					}
+					markedw &= markedw - 1
+				}
+				b = b.next.Load()
+				if b == nil {
+					if loadExitFlag == -1 {
+						return *new(V), false
+					}
+					break load
+				}
+			}
+		}
+
 		rootb.mu.Lock()
```

Comment:
`loadExitFlag` should be checked in the later code as at the point when we acquired the mutex, the key may be already present in the map.